### PR TITLE
Userlog init PR

### LINKED ISSUE #49: Create main Features (without review)
- [x] Create a main screen feature
- [x] Create a button on main screen
- [x] Create About screen feature
- [x] Open About by clicking on the button on main screen

https://github.com/sahlet-official/buttonlog/tree/develop/Specification/UI

______________

This issue size:
- Estimate SP: 5
- Actual SP: 3

### PR DESCRIPTION
**Pull request by Yura-0**
Resolve #49

### CODE REVIEW

File: Sources/Userlog/user_log/lib/features/about_screen_feature/about_screen.dart [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import 'package:flutter/material.dart';
```
Comment (sahlet-official) [inline Sources/Userlog/user_log/lib/features/about_screen_feature/about_screen.dart:1]: Make folder structures inside features

File: Sources/Userlog/user_log/lib/features/about_screen_feature/about_screen.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:user_log/core/assets.dart';
```
Comment (sahlet-official) [inline Sources/Userlog/user_log/lib/features/about_screen_feature/about_screen.dart:3]: Search for examples of how to use
Injectable
get_it
BLOC

And use in our project

File: Sources/Userlog/user_log/lib/features/about_screen_feature/about_screen.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import 'package:user_log/core/fonts.dart';
+import 'package:user_log/core/texts.dart';
+
```
Comment (sahlet-official) [inline Sources/Userlog/user_log/lib/features/about_screen_feature/about_screen.dart:6]: Make api folder in features where it needed

**Review by sahlet-official — CHANGES_REQUESTED** [review]

File: Sources/Userlog/userlog/lib/features/about_screen_feature/assets/about_images.dart [hunk 4]
```diff
@@ -0,0 +1,3 @@
+abstract class AboutImages {
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/features/about_screen_feature/assets/about_images.dart:1]: Rename features without "screen" and "feature" words

File: Sources/Userlog/userlog/lib/features/about_screen_feature/assets/about_screen_texts.dart [hunk 5]
```diff
@@ -0,0 +1,5 @@
+abstract class AboutTexts {
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/features/about_screen_feature/assets/about_screen_texts.dart:1]: remove the word screen from the name of the file. The name of file has to correspond to the name of the class

File: Sources/Userlog/userlog/lib/features/about_screen_feature/screens/about_screen.dart [hunk 6]
```diff
@@ -0,0 +1,98 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:userlog/common/styles/fonts.dart';
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/features/about_screen_feature/screens/about_screen.dart:3]: Make spaces between categories of imports

File: Sources/Userlog/userlog/lib/features/main_screen_feature/screens/main_screen.dart [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+                IconButton(
+                  icon: SvgPicture.asset(
+                    MainImages.infoButtonImage,
+                    fit: BoxFit.cover,
+                  ),
+                  iconSize: 33,
+                  onPressed: () {
+                    Navigator.pushNamed(
+                      context,
+                      '/About',
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/features/main_screen_feature/screens/main_screen.dart:174]: That is why we need to make api folder in every feature.
This route has to be exported from about feature as api

**Review by sahlet-official — CHANGES_REQUESTED** [review]

File: Sources/Userlog/userlog/lib/features/about/api/about_feature.dart [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/widgets.dart';
+
+import 'package:userlog/common/infrastructure/feature_base/feature_api.dart';
+import 'package:userlog/features/about/api/about_routes.dart';
+import 'package:userlog/features/about/screens/about_screen.dart';
+
+class AboutApi extends FeatureApi {
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/features/about/api/about_feature.dart:7]: Lets rename AboutApi to AboutFeature

File: Sources/Userlog/userlog/lib/features/main/api/main_feature.dart [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import 'package:flutter/widgets.dart';
+
+import 'package:userlog/common/infrastructure/feature_base/feature_api.dart';
+import 'package:userlog/features/main/api/main_routes.dart';
+import 'package:userlog/features/main/screens/clear_log_confirmation_screen.dart';
+import 'package:userlog/features/main/screens/main_screen.dart';
+
+class MainApi extends FeatureApi {
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/features/main/api/main_feature.dart:8]: Let's rename MainApi to MainFeature

File: Sources/Userlog/userlog/lib/main.dart [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+class UserLogApp extends StatelessWidget {
+  const UserLogApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    List<FeatureApi> featuresApi = [MainApi(), AboutApi()];
+    Map<String, WidgetBuilder> routes = {};
+    for (var e in featuresApi) {
+      routes.addAll(e.getRoutes());
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:19]: Will "addAll" throw an exception if there will be a repeated route?

File: Sources/Userlog/userlog/lib/main.dart [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    return MaterialApp(
+      title: 'UserLog',
+      theme: ThemeData(
+        backgroundColor: Colors.white,
+      ),
+      initialRoute: '/',
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:27]: replace '/' to MainRoutes.root

File: Sources/Userlog/userlog/lib/main.dart (hunk 10 above)
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:19]: call these functions (addAll and getRoutes) in different lines.

As easier code - as better

File: Sources/Userlog/userlog/lib/main.dart [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+void main() {
+  runApp(const UserLogApp());
+}
+
+class UserLogApp extends StatelessWidget {
+  const UserLogApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    List<FeatureApi> featuresApi = [MainApi(), AboutApi()];
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:16]: write features names classes in different lines (make in as column)

**Review by sahlet-official — CHANGES_REQUESTED** [review]

File: Sources/Userlog/userlog/lib/main.dart (hunk 10 above)
Comment (Yura-0) [reply Sources/Userlog/userlog/lib/main.dart:19]: No

Comment (Yura-0) [reply Sources/Userlog/userlog/lib/main.dart:19]: To do this, I would need to store the value of e.getRoutes into a variable, which would force the creation of that variable, which would make the operation more complicated. Or do you have something else requesting this change

File: Sources/Userlog/userlog/lib/main.dart [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+      }
+      routes[key] = route[key]!;
+    }
+
+    return routes;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<UiFeature> featuresApi = [MainFeature(), AboutFeature()];
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:29]: write it like that:
........... = [
MainFeature,
AboutFeature,
]

Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:29]: rename featuresApi to features

**Review by sahlet-official — CHANGES_REQUESTED** [review]

File: Sources/Userlog/userlog/lib/main.dart (hunk 13 above)
Comment (Yura-0) [reply Sources/Userlog/userlog/lib/main.dart:29]: I did that, but when I auto-formatted the code, it reverted the array to that look again

Comment (sahlet-official) [reply Sources/Userlog/userlog/lib/main.dart:29]: @Yura-0 
Ok, leave like that

File: Sources/Userlog/userlog/lib/main.dart [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+import 'package:userlog/features/main/api/main_routes.dart';
+
+void main() {
+  runApp(const UserLogApp());
+}
+
+class UserLogApp extends StatelessWidget {
+  const UserLogApp({super.key});
+
+  Map<String, WidgetBuilder> _isSameRoute(
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:15]: It is a very redundant code.

make a function that gets features as parameter and returns routes

**Review by sahlet-official — CHANGES_REQUESTED** [review]

File: Sources/Userlog/userlog/lib/main.dart [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+class UserLogApp extends StatelessWidget {
+  const UserLogApp({super.key});
+
+  Map<String, WidgetBuilder> _getRoutes(List<UIFeature> features) {
+    Map<String, WidgetBuilder> route;
+    Map<String, WidgetBuilder> routes = {};
+    for (var e in features) {
+      route = e.getRoutes();
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:19]: change it to
Map<String, WidgetBuilder> featureRoutes = e.getRoutes();

File: Sources/Userlog/userlog/lib/main.dart [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+      for (var key in route.keys) {
+        if (routes.containsKey(key)) {
+          throw "This route already exists";
+        }
+        routes.addAll(route);
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/main.dart:24]: this line should be out of this cycle.

**Review by sahlet-official — CHANGES_REQUESTED** [review]

**Review by sahlet-official — DISMISSED** [review]

**Comment by sahlet-official** [comment]
Аппрув

**Pull request merged**